go.mod

```diff
@@ -6,6 +6,7 @@ require (
 	github.com/cert-manager/cert-manager v1.9.1
 	github.com/go-bindata/go-bindata v3.1.2+incompatible
 	github.com/google/go-jsonnet v0.17.0
+	github.com/imdario/mergo v0.3.12
```

Comment:
I would prefer a solution without relying on external libraries. If logic https://github.com/openshift/library-go/blob/master/pkg/operator/unsupportedconfigoverridescontroller/unsupportedconfigoverrides_controller.go here doesn't solve our issue, then maybe we could look at external libraries. WDYT?

Reply:
As discussed, I've updated the implementation to not use mergo, as it isn't much of a help for the given use case. 
You could PTAL the latest changes.